### PR TITLE
Go: Storage Rules

### PR DESCRIPTION
**Pull request by hiteshbedre**

### CODE REVIEW

File: rules/sinks/storages/neo4j/go.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# Sink Rules for storage database Neo4j Graph Database - https://neo4j.com/
+
+sinks:
+
+  - id: Storages.Neo4jGraphDatabase.ReadAndWrite
+    name: Neo4j Graph Database(ReadAndWrite)
+    domains:
+      - neo4j.com
+    patterns:
+      - "(?i)(github.com/neo4j/neo4j-go-driver).*(ExecuteQuery|ExecuteRead)"
```
Comment (khemrajrathore) [inline rules/sinks/storages/neo4j/go.yaml:11]: Execute Query makes sense for Read and write, but `ExecuteRead` looks like is a read operation

File: rules/sinks/storages/sqlite/go.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Sink Rules for storage database Android SQLite - https://developer.android.com/
+
+sinks:
+
+  - id: Storages.Android.SQLite.Read
+    name: SQLite(Read)
+    domains:
+      - android.com
+    patterns: 
+      - "(?i)(github.com/mattn/go-sqlite).*(Query|QueryRow)"
```
Comment (khemrajrathore) [inline rules/sinks/storages/sqlite/go.yaml:11]: Query and QueryRow looks likea ReadAndWrite operation

File: rules/sinks/storages/weaviate/go.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# Sink Rules for storage database Vaticle TypeDB - https://vaticle.com/typedb
+
+sinks:
+
+  - id: Storages.Weaviate.ReadAndWrite
+    name: Weaviate VectorDB
+    domains:
+      - weaviate.io
+    patterns:
+      - "(?i)(github.com/semi-technologies/weaviate-go-client).*(ObjectsBatcher|Get|GetSearch)"
```
Comment (khemrajrathore) [inline rules/sinks/storages/weaviate/go.yaml:11]: Get, GetSearch looks like a Read operation

**Comment by khemrajrathore** [comment]
@hiteshbedre For most of the rules I see only `ReadAndWrite` kind of rule, Have we not come across there separate `Read` or `Write` operations?

File: rules/sinks/storages/sqlite/go.yaml (hunk 2 above)
Comment (hiteshbedre) [reply rules/sinks/storages/sqlite/go.yaml:11]: Checked again, query and QueryRow operation in SQLlite are used as read.

File: rules/sinks/storages/neo4j/go.yaml (hunk 1 above)
Comment (hiteshbedre) [reply rules/sinks/storages/neo4j/go.yaml]: Addressed here: https://github.com/Privado-Inc/privado/pull/317/commits/8d15ec0632dcb4ffc46f69741c7fea037b25da44

File: rules/sinks/storages/weaviate/go.yaml (hunk 3 above)
Comment (hiteshbedre) [reply rules/sinks/storages/weaviate/go.yaml:11]: addressed here: https://github.com/Privado-Inc/privado/pull/317/commits/8d15ec0632dcb4ffc46f69741c7fea037b25da44

File: rules/sinks/storages/weaviate/go.yaml [hunk 4]
```diff
@@ -10,3 +10,11 @@ sinks:
     patterns:
       - "(?i)(github.com/semi-technologies/weaviate-go-client).*(ObjectsBatcher|Get|GetSearch)"
```
Comment (khemrajrathore) [inline rules/sinks/storages/weaviate/go.yaml:11]: remove Get|GetSearch from ReadAndWrite rule

Comment (hiteshbedre) [reply rules/sinks/storages/weaviate/go.yaml]: Addressed here: https://github.com/Privado-Inc/privado/pull/317/commits/264cd85ba5652b39c583db11488a8f8a908fb21a

**Pull request merged**